tests/integration/netdev.bats

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	requires root
+	setup_busybox
+
+	# skip to setup resources that require root
+	# to avoid the test to fail
+	if [ $EUID -ne 0 ] || in_userns; then
+		# create a dummy interface to move to the container
+		ip link add dummy0 type dummy
+
+		# create a temporary name for the test network namespace
+		tmp=$(mktemp)
+		rm -f "$tmp"
+		ns_name=$(basename "$tmp")
+
+		# create network namespace
+		ip netns add "$ns_name"
+		ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+	fi
+}
+
+function teardown() {
+	if [ $EUID -ne 0 ] || in_userns; then
+		ip netns del "$ns_name"
+		ip link del dev dummy0
+	fi
+
+	teardown_bundle
+}
+
+@test "move network device to container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+      		| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to container network namespace and restore it back" {
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	update_config ' .linux.netDevices |= {"dummy0": {} }'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	# the network namespace owner controls the lifecycle of the interface
+	ip netns exec "$ns_name" ip link set dev dummy0 netns 1
```

Comment:
Is number 1 a magic number that refers to the initial netns, then?

Can you change the comment to explain what 1 is and say we move it back to the initial netns?

Reply:
added